src/Bridge/Doctrine/Orm/Extension/FilterExtension.php

```diff
@@ -47,7 +48,7 @@ public function applyToCollection(QueryBuilder $queryBuilder, string $resourceCl
 
         foreach ($this->filters as $filterName => $filter) {
             if (in_array($filterName, $resourceFilters) && $filter instanceof FilterInterface) {
-                $filter->apply($queryBuilder, $resourceClass, $operationName);
+                $filter->apply($queryBuilder, new QueryNameGenerator(), $resourceClass, $operationName);
```

Comment:
`QueryNameGenerator` should be instantiated in `ItemDataProvider`/`CollectionDataProvider` to avoid name collisions.